pyresample/_spatial_mp.py

```diff
@@ -112,6 +112,11 @@ def query(self, x, k=1, eps=0, p=2, distance_upper_bound=np.inf):
 class BaseProj(pyproj.Proj):
 
     def __init__(self, projparams=None, preserve_units=True, **kwargs):
+        # Copy dict-type arguments as they will be modified in-place
+        kwargs = kwargs.copy()
```

Comment:
Why is this copy needed? The `**kwargs` should already be a copy of whatever dictionary/keywords were passed.

Reply:
Oh, good point. Only `projparams` need to be copied then